integration-tests/suites/async_connections.go

```diff
@@ -0,0 +1,96 @@
+
+import (
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/stackrox/collector/integration-tests/suites/common"
+	"github.com/stackrox/collector/integration-tests/suites/config"
+)
+
+type AsyncConnectionTestSuite struct {
+	IntegrationTestSuiteBase
+	BlockConnection bool
+
+	clientContainer string
+	serverContainer string
+	serverIP        string
+}
+
+/*
+ * This suite has 2 main cases:
+ * - BlockConnection=false: verify that a successful asynchronous connection gets reported
+ *     We provide curl-test with the IP of a TCP server to isntruct it to connect to this server.
+ * - BlockConnection=true : verify that a failed aynchronous connection is not
+ *     No parameter to curl-test, and in this case, it simulates a firewall rule by connecting to
+ *     a dummy address.
+ */
+func (s *AsyncConnectionTestSuite) SetupSuite() {
+
+	s.metrics = map[string]float64{}
+	s.executor = common.NewExecutor()
+	s.StartContainerStats()
+	s.collector = common.NewCollectorManager(s.executor, s.T().Name())
+
+	// we need a short scrape interval to have it trigger after the connection is closed.
+	s.collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":false,"scrapeInterval":2}`
+
+	err := s.collector.Setup()
+	s.Require().NoError(err)
+
+	err = s.collector.Launch()
+	s.Require().NoError(err)
+	time.Sleep(30 * time.Second) // TODO use readiness
+
+	image_store := config.Images()
+
+	containerID, err := s.launchContainer("server", image_store.ImageByKey("nginx"))
+	s.Require().NoError(err)
+	s.serverContainer = common.ContainerShortID(containerID)
+
+	s.serverIP, err = s.getIPAddress("server")
+	s.Require().NoError(err)
+
+	time.Sleep(5 * time.Second) // TODO use the endpoint declaration
+
+	if !s.BlockConnection {
+		containerID, err = s.launchContainer("client", image_store.QaImageByKey("qa-async-connection"), s.serverIP)
+	} else {
+		containerID, err = s.launchContainer("client", image_store.QaImageByKey("qa-async-connection"))
+	}
+	s.Require().NoError(err)
+	s.clientContainer = common.ContainerShortID(containerID)
+
+	time.Sleep(10 * time.Second) // give some time to the connection to fail
+
+	err = s.collector.TearDown()
+	s.Require().NoError(err)
+
+	s.db, err = s.collector.BoltDB()
+	s.Require().NoError(err)
+}
+
+func (s *AsyncConnectionTestSuite) TearDownSuite() {
+	s.cleanupContainer([]string{"server", "client"})
+	stats := s.GetContainerStats()
+	s.PrintContainerStats(stats)
+	s.WritePerfResults("AsyncConnection", stats, s.metrics)
+}
+
+func (s *AsyncConnectionTestSuite) TestNetworkFlows() {
+
+	if !s.BlockConnection {
+		// expect one connection
+		networkInfos, err := s.GetNetworks(s.clientContainer)
+		s.Require().NoError(err)
+
+		assert.Equal(s.T(), 1, len(networkInfos))
+	} else {
+		// expect no connections at all
+		hasNetwork, err := s.HasNetworks(s.clientContainer)
+		s.Require().NoError(err)
+
+		assert.True(s.T(), !hasNetwork, "A failed asynchronous connection should not be reported")
+	}
```

Comment:
```suggestion
        networkInfos, err := s.GetNetworks(s.clientContainer)
	if !s.BlockConnection {
		// expect one connection		
		s.Require().NoError(err)

		assert.Equal(s.T(), 1, len(networkInfos))
	} else {
		// expect no connections at all
		s.Require().Error(err)
	}
```

Reply:
Well, that could be any kind of error. Overall, not receiving networks does not seem like an error to me, especially when you expect that there will be no networks reported.

Reply:
Implemented as suggested.

Reply:
Maybe you could check for the specific error? Maybe in a different PR we could make it so that not finding the bucket is not an error. I will approve as is. There is another unrelated comment that I left that I would like to be resolved before merging.

Reply:
I made an attempt to return empty set; but expectations that the set is empty or not-empty are rather hard to read. I believe that if we do this in a subsequent PR, it would still be beneficial if we have functions that expect empty or something, instead of expect(x.len() > 0)